aws/ami.go

```diff
@@ -35,14 +36,30 @@ func getAllAMIs(session *session.Session, region string, excludeAfter time.Time)
 			return nil, err
 		}
 
-		if excludeAfter.After(createdTime) {
+		if excludeAfter.After(createdTime) && !ImageHasAWSBackupTag(image.Tags) {
 			imageIds = append(imageIds, image.ImageId)
 		}
 	}
 
 	return imageIds, nil
 }
 
+// Check if the image has an AWS Backup tag
+// Resources created by AWS Backup are listed as owned by self, but are actually
+// AWS managed resources and cannot be deleted here.
+func ImageHasAWSBackupTag(tags []*ec2.Tag) bool {
```

Comment:
I think this might be applicable beyond just AMIs, do we have a generic place we can put this so it's easy to find for other consumers?

Reply:
I suppose we could home it in go-commons and load it from there?

Reply:
I think it makes sense in this package specifically, i'm just wondering if we should put it in `helpers.go` or something like that?

Reply:
@arsci  I know that the EBS snapshots that back the AMIs get a similar tag.

I'd imagine the behavior would be similar for the AWS services listed here: https://docs.aws.amazon.com/aws-backup/latest/devguide/whatisbackup.html#supported-resources Looking at that list I think I'd probably be most immediately concerned by RDS - e.g., if someone was using backup to manage RDS snapshots -> tried to run cloud nuke they might experience the same problem we're solving for here.

That being said, I think for now this is probably fine and we can circle up internally on how to verify the behavior for other resources.